ilastik/workflows/neuralNetwork/_nnWorkflowBase.py

```diff
@@ -166,14 +166,11 @@ def handleAppletStateUpdateRequested(self, upstream_ready=True):
         input_ready = len(opDataSelection.ImageGroup) > 0 and not self.dataSelectionApplet.busy
```

Comment:
It would be nice to group op shortcuts, flags, and applet changes separately, and also replace `busy` with `ready` where possible.

```python
opDataSelection = self.dataSelectionApplet.topLevelOperator
opNNClassification = self.nnClassificationApplet.topLevelOperator
opDataExport = self.dataExportApplet.topLevelOperator

data_selection_ready = not self.dataSelectionApplet.busy
input_ready = data_selection_ready and len(opDataSelection.ImageGroup) > 0
nn_ready = input_ready and opNNClassification.ModelSession.ready()
predictions_ready = nn_ready and len(opDataExport.Inputs) > 0
batch_processing_ready = not self.batchProcessingApplet.busy

self._shell.setAppletEnabled(..., upstream_ready and batch_processing_ready)
...
```

Maybe it would also make sense to merge `batch_processing_ready` into `upstream_ready`, but I'm not sure about this:
```python
upstream_ready &= not self.batchProcessingApplet.busy
```